runtime/near-vm-runner/src/near_vm_runner.rs

```diff
@@ -249,16 +249,17 @@ impl NearVM {
                 // FIXME: should have as many code memories as there are possible parallel
                 // invocations of the runtime… How do we determine that? Should we make it
                 // configurable for the node operators, perhaps, so that they can make an informed
-                // choice based on the amount of memory they have and shards they track?
+                // choice based on the amount of memory they have and shards they track? Should we
+                // actually use some sort of semaphore to enforce a parallelism limit?
                 //
-                // NB: 16MiB is a best guess as to what the maximum size a loaded artifact can
+                // NB: 64MiB is a best guess as to what the maximum size a loaded artifact can
                 // plausibly be. This is not necessarily true – there may be WebAssembly
                 // instructions that expand by more than 4 times in terms of instruction size after
                 // a conversion to x86_64, In that case a re-allocation will occur and executing
                 // that particular function call will be slower. Not to mention there isn't a
                 // strong guarantee on the upper bound of the memory that the contract runtime may
                 // require.
-                LimitedMemoryPool::new(32, 16 * 1024 * 1024).unwrap_or_else(|e| {
+                LimitedMemoryPool::new(8, 64 * 1024 * 1024).unwrap_or_else(|e| {
```

Comment:
Especially with 64MiB mmaps, it might make sense to poke each page of each mapping to make sure the kernel didn’t overcommit them. It’d make initial startup a tad slower, but we’d avoid the variance of pagefaulting and needing to reallocate

Reply:
Faulting in maps like this is somewhat of a bogus thing to do. Kernel can freely swap out pages from memory even after they have been faulted in. So in a memory-starved situations the same cliff still exists. The work-around to this behaviour is to use `mlock`, however using that opens its own can of worms – systems aren’t typically set up for a huge amount of locked memory. For instance on my system 8MiB is the maximum you can `mlock` without further (`rlimit`) changes.

Reply:
Good point, I had thought of the kernel just not allocating the memory, not of the kernel swapping it out.